posts/2023-09-26-23.0.0.10-beta.adoc

```diff
@@ -0,0 +1,628 @@
++
+[source,xml]
+----
+<ltpa monitorDirectory="true" monitorInterval="5"/>
+----
++
+The `monitorDirectory` attribute monitors the `${server.config.dir}/resources/security/` directory by default, but can monitor any directory the primary keys file is specified in. The directory monitor looks for any LTPA keys files with the `.keys` extension. The Open Liberty server reads these LTPA keys and uses them as Validation Keys.
++  
+If the `monitorInterval` is set to `0`, the default value, the directory stops being monitored.
++
+The `ltpa.keys` file can be renamed, for example, `validation1.keys` and then the Liberty server automatically regenerates a new `ltpa.keys` file with new primary keys that are used for all new LTPA tokens created. The keys in `validation1.keys` continue to be used for validating existing LTPA tokens. 
++   
+When the `validation1.keys` are no longer needed, remove them from usage by deleting the file, or by setting `monitorDirectory` to `false`. It is recommended to remove validation keys as it can improve performance.
+
+* Specify the validation keys file and optionally specify a date-time to stop using the validation keys.
+   1. Copy the primary keys file (`ltpa.keys`) to a validation keys file, for example `validation1.keys`.
+   2. Modify the server configuration to use the validation keys file  by specifying a `<validationKeys>` server configuration element inside the `<ltpa>` element. For example, add the following configuration to the `server.xml` file:
++
+[source,xml]
+----
+<ltpa>
+    <validationKeys fileName="validation1.keys" password="{xor}Lz4sLCgwLTs=" notUseAfterDate="2024-01-02T12:30:00Z"/>
+<ltpa/>  
+----
++
+The `validation1.keys` file can be removed from usage at a specified date-time in the future with the optional attribute, `notUseAfterDate`.  It is recommended to use `notUseAfterDate` to remove validation keys after a given period as it can improve performance.
++
+The `fileName` and `password` attributes are required in the `validationKeys` element, but `notUseAfterDate` is optional.
++
+After the validation keys file is loaded from the server configuration update, the original primary keys file (`ltpa.keys`) can be deleted, which triggers new primary keys to be created while continuing to use `validation1.keys` for validation.
++
+Specifying validation keys this way can also be combined with using monitor directory to ready more validation keys that are specified in server.xml configuration. For example:
++
+[source,xml]
+----
+<ltpa monitorDirectory="true" monitorInterval="5m">
+    <validationKeys fileName="validation1.keys" password="{xor}Lz4sLCgwLTs=" notUseAfterDate="2024-01-02T12:30:00Z"/>
+<ltpa/>
+----
++     
+To see all of the Liberty `<ltpa>` server configuration options see link:/docs/latest/reference/config/ltpa.html[LTPA configuration docs].
+    
+    

```

Comment:
another paragraph I was asked to add to the end:
```suggestion
  This feature also provides the new capability to perform one-way Single Sign-On (SSO) for LTPA tokens which was not previously possible. Before this feature, SSO for LTPA tokens could only be bi-directional, not one-way, and required using the same `ltpa.keys` file on every applicable Liberty server. The new "one-way" SSO  means a user will be able to use SSO from Liberty server1 to server2, but not from server2 to server1.
 - If you have two (or more) Liberty servers, for example server1 and server2, each with a different LTPA primary keys file - Adding the primary keys file from server1, as a validation keys file to server2 will allow one-way single sign-on capability from server1 to server2. 
  
  - A user who authenticated to server1 first will be able to access server2 with SSO. However, a user who authenticates to server2 first will need to re-authenticate again before accessing server1 because server1 does not have the primary keys file from server2.

```

Reply:
Will leave this out (as discussed on Slack) because this is useful in traditional deployments but not relevant to cloud-native deployments, which is our focus here.